hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -4004,12 +4004,14 @@ public void cleanUp() {
      * Background action to act on references being removed.
      */
     private static class StatisticsDataReferenceCleaner implements Runnable {
+      private static int REF_QUEUE_POLL_TIMEOUT = 100;
```

Comment:
that's going to be waking every 100 milliseconds, demanding cpu time etc etc. If there has to be a timeout, it needs to be something less disruptive, like 100 seconds.

What would happen if that was the case?